src/analysis/retail/priest/holy/modules/core/HolyWordCDRBySpell.tsx

```diff
@@ -0,0 +1,225 @@
+          cdrFromBase: baseCd,
+          cdrFromLOTN: cdrFromLotn,
+          cdrFromApoth: cdrFromApoth,
+          cdrFromTwwTier: cdrFromTwwTier,
+          cdrFromVoh: vohCD,
+          spellNum: spellId,
+          totalCDR: totalEffectiveCDR,
+          affectedSpell: affectedSpell,
+        };
+      }
+    }
+  }
+
+  /**
+   *
+   * SPECIAL CASES
+   *
+   */
+
+  handleEnergyCycle(event: RemoveBuffEvent | RemoveBuffStackEvent) {
+    // this uses heal event because tww S1 tier set is stupid and doesn't "cast" it just procs a second heal
+    // but still consumes SoL potentially
+    const castEvent = getSOLFlashCast(event);
+
+    if (castEvent) {
+      if (buffedBySurgeOfLight(event)) {
+        this.handleOnCast(castEvent, 'ENERGY_CYCLE');
+      }
+    }
+  }
+
+  statistic() {
+    this.hwContainer = this.hwContainer.filter(Boolean);
+    this.hwContainer.sort((a, b) => {
+      if (a.affectedSpell < b.affectedSpell) {
+        return -1;
+      }
+      if (a.affectedSpell > b.affectedSpell) {
+        return 1;
+      }
+      return 0;
+    });
+    return (
+      <Statistic
+        position={STATISTIC_ORDER.CORE(1)}
+        size="flexible"
+        wide
+        dropdown={
+          <>
+            <table className="table table-condensed">
+              <thead>
+                <tr>
+                  <td className="text-center">Cast Spell</td>
+                  <td className="text-center">Reduced Spell</td>
+                  <td>Base</td>
+                  {this.apotheosisActive && (
+                    <th>
+                      <SpellIcon spell={TALENTS.APOTHEOSIS_TALENT}></SpellIcon>
+                    </th>
+                  )}
+                  {this.lightOfTheNaaruActive && (
+                    <th>
+                      <SpellIcon spell={TALENTS.LIGHT_OF_THE_NAARU_TALENT}></SpellIcon>
+                    </th>
+                  )}
+                  {this.voiceOfHarmonyActive && (
+                    <th>
+                      <SpellIcon spell={TALENTS.VOICE_OF_HARMONY_TALENT}></SpellIcon>
+                    </th>
+                  )}
+                  {this.tierActive && <th> Tier</th>}
+                  <th> Total Used</th>
+                  <th> Wasted</th>
+                </tr>
+              </thead>
+              <tbody>
+                {Array.from(this.hwContainer.keys()).map((e, i) => (
+                  <tr key={Number(i)}>
+                    <td className="text-center">
+                      <SpellIcon spell={this.hwContainer[Number(e)].spellNum || 2060} />
+                    </td>
+                    <td className="text-center">
+                      <SpellIcon spell={this.hwContainer[Number(e)].affectedSpell || 2060} />
+                    </td>
+                    <td>{Math.round(this.hwContainer[e].cdrFromBase * 10) / 10}s</td>
+                    {this.apotheosisActive && (
+                      <td>{Math.round(this.hwContainer[e].cdrFromApoth * 10) / 10}s</td>
```

Comment:
Could be nice to put the round formula into a helper function to reduce code duplication and also help readers understand what you're doing.